server/memcached/src/main/java/org/infinispan/server/memcached/MemcachedDecoder.java

```diff
@@ -296,6 +301,26 @@ private void decodeValue(ChannelHandlerContext ctx, ByteBuf buffer, MemcachedDec
       writeResponse(ch, ret);
    }
 
+   private Object touch() {
+      final CacheEntry<byte[], byte[]> cacheEntry = cache.getCacheEntry(key);
+      if (cacheEntry == null) {
+         return createNotExistResponse();
+      }
+      final Metadata newMetadata = new MemcachedMetadata.Builder()
+              .merge(cacheEntry.getMetadata())
+              .lifespan(params.lifespan > 0 ? toMillis(params.lifespan) : -1)
+              .build();
+      cache.replace(key, cacheEntry.getValue(), newMetadata);
```

Comment:
It is unclear from https://github.com/memcached/memcached/blob/master/doc/protocol.txt but we technically have 2 expiration types, lifespan and max idle.  It sounds like memcached uses the latter only. In this case we don't need to do a replace, but rather just the get alone would be sufficient to refresh the max idle time.

Reply:
so infinispan is simply ignoring the exptime from the memcached protocol and also ignores the lifespan stored in the cache entry's metadata?
what else is the metadata->lifespan property used for and why is it stored with the key/value entry?

Reply:
Sorry, I am rereading the protocol txt file you linked in the JIRA. It sounds like emulating the guarantees they state would be best using lifespan, and ignore max-idle as our put and your new code is doing.

Reply:
so I did everything right by making a replace with new metadata (and lifespan)?

Reply:
I am not sure what the semantics are around a concurrent touch with another write operation. But I would say we probably don't have to worry about it, as they are most likely undefined.